modules/imap/hm-imap.php

```diff
@@ -761,7 +761,7 @@ public function poll() {
          * @param bool $raw flag to disable decoding header values
          * @return array list of headers and values for the specified uids
          */
-        public function get_message_list($uids, $raw=false) {
+        public function get_message_list($uids, $raw=false, $include_preview = false) {
```

Comment:
Implementation is much better than previous time.  Please respect "include_preview" option - i.e. don't fetch BODY below if preview is not requested. Add a user config setting to fetch previews - boolean - yes/no. Pass it to this method, so unless user wants previews, it won't fetch the body.

Reply:
well noted